public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx

```diff
@@ -12,8 +12,21 @@ import { LokiOptionFields } from './LokiOptionFields';
 type Props = QueryEditorProps<LokiDatasource, LokiQuery, LokiOptions>;
 
 export function LokiQueryEditor(props: Props) {
-  const { range, query, data, datasource, onChange, onRunQuery } = props;
-  const absoluteTimeRange = { from: range!.from!.valueOf(), to: range!.to!.valueOf() }; // Range here is never optional
+  const { app, query, data, datasource, onChange, onRunQuery } = props;
+
+  if (app === CoreApp.CloudAlerting) {
+    return (
+      <LokiQueryField
```

Comment:
just wondering, why not use `<LokiQueryField` directly and not modify `<LokiQueryEditor` ? would not need the `app` then

Reply:
ah, I guess so we can have abstraction over `QueryEditor`

Reply:
@torkelo ☝🏻

Reply:
Many data sources will probably want some options to be hidden or shown for alerting queries, this way this is possible and not a specific thing for loki & prometheus.